src/RenderWebGL.js

```diff
@@ -1002,10 +1002,19 @@ class RenderWebGL extends EventEmitter {
      * RenderConstants.ID_NONE if there is no Drawable at that location.
      */
     pick (centerX, centerY, touchWidth, touchHeight, candidateIDs) {
+        const bounds = this.clientSpaceToScratchBounds(centerX, centerY, touchWidth, touchHeight);
+
         candidateIDs = (candidateIDs || this._drawList).filter(id => {
             const drawable = this._allDrawables[id];
+            const drawableBounds = this.getBounds(id);
+            const inRange = (
+                bounds.top < drawableBounds.top
+                    && bounds.right < drawableBounds.right
+                    && bounds.bottom > drawableBounds.bottom
+                    && bounds.left > drawableBounds.left
+            );
             // default pick list ignores visible and ghosted sprites.
-            if (drawable.getVisible() && drawable.getUniforms().u_ghost !== 0) {
+            if (inRange && drawable.getVisible() && drawable.getUniforms().u_ghost !== 0) {
```

Comment:
You could move the bounds check in here, like so:

```
if (drawable.getVisible() && drawable.getUniforms().u_ghost !== 0) {
    const drawableBounds = drawable.getFastBounds();
    const inRange = bounds.intersects(drawableBounds);
    if (!inRange) return false;
    // then update the matrix and silhouette...
```
This avoids getting the bounds (and hence updating the transform matrix) for drawables which aren't visible anyway.